evalml/tests/component_tests/test_varmax_regressor.py

```diff
@@ -0,0 +1,393 @@
+        (False, True, False, False, False, True),
+        (False, False, True, False, True, False),
+        (False, False, True, True, False, False),
+        (False, True, True, True, False, False),
+        (False, True, True, False, True, False),
+    ],
+)
+def test_fit_predict_sk_failure(
+    train_features_index_dt,
+    train_target_index_dt,
+    train_none,
+    no_features,
+    datetime_feature,
+    test_features_index_dt,
+    ts_multiseries_data,
+):
+    from sktime.forecasting.varmax import VARMAX
+
+    X_train, X_test, y_train = ts_multiseries_data(
+        train_features_index_dt,
+        train_target_index_dt,
+        train_none,
+        datetime_feature,
+        no_features,
+        test_features_index_dt,
+        match_indices=False,
+    )
+
+    a_clf = VARMAX(maxiter=10)
+    with pytest.raises(Exception):
+        a_clf.fit(X=X_train, y=y_train)
+
+    m_clf = VARMAXRegressor(series_id="series_id")
+    m_clf.fit(X=X_train, y=y_train)
+    y_pred = m_clf.predict(X=X_test)
+    assert isinstance(y_pred, pd.DataFrame)
+    assert len(y_pred) == 10
+    assert y_pred.index.equals(X_test.index)
+
+
+@pytest.mark.parametrize("freq_num", ["1", "2"])
+@pytest.mark.parametrize("freq_str", ["T", "M", "Y"])
+def test_different_time_units_out_of_sample(
+    freq_str,
+    freq_num,
+    sktime_varmax,
+    forecasting,
+):
+    from sktime.forecasting.base import ForecastingHorizon
+    from sktime.forecasting.varmax import VARMAX
+
+    datetime_ = pd.date_range("1/1/1870", periods=20, freq=freq_num + freq_str)
+
+    X = pd.DataFrame(
+        range(20),
+        index=datetime_,
+    )
+    y = pd.DataFrame(
+        {
+            "target_0": pd.Series(
+                np.sin(np.linspace(-8 * np.pi, 8 * np.pi, 20)),
+                index=datetime_,
+            ),
+            "target_1": pd.Series(
+                np.sin(np.linspace(-4 * np.pi, 4 * np.pi, 20)),
+                index=datetime_,
+            ),
+        },
+    )
+
+    fh_ = ForecastingHorizon([i + 1 for i in range(len(y[15:]))], is_relative=True)
+
+    a_clf = VARMAX(maxiter=10)
+    clf = a_clf.fit(X=X[:15], y=y[:15])
+    y_pred_sk = clf.predict(fh=fh_, X=X[15:])
+
+    m_clf = VARMAXRegressor(series_id="series_id")
+    m_clf.fit(X=X[:15], y=y[:15])
+    y_pred = m_clf.predict(X=X[15:])
+
+    np.testing.assert_almost_equal(y_pred_sk.values, y_pred.values)
+    assert y_pred.index.equals(X[15:].index)
+
+
+@patch("sktime.forecasting.varmax.VARMAX.fit")
+def test_varmax_supports_boolean_features(mock_fit):
+    X = pd.DataFrame({"dates": pd.date_range("2021-01-01", periods=10)})
+    X.ww.init()
+    X.ww["bool_1"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=0)
+        .reset_index(drop=True)
+    )
+    X.ww["bool_2"] = (
+        pd.Series([True, False])
+        .sample(n=10, replace=True, random_state=1)
+        .reset_index(drop=True)
+    )
```

Comment:
Small question - why set up X this way, rather than create a dataframe with all three columns to begin with?